velodyne_pointcloud/include/velodyne_pointcloud/point_types.h

```diff
@@ -0,0 +1,48 @@
+/* -*- mode: C++ -*-
+ *
+ *  Copyright (C) 2011, 2012 Austin Robot Technology
+ *
+ *  License: Modified BSD Software License Agreement
+ *
+ *  $Id: data_base.h 1554 2011-06-14 22:11:17Z jack.oquin $
+ */
+
+/** \file
+ *
+ *  Point Cloud Library point structures for Velodyne data.
+ *
+ *  @author Jesse Vera
+ *  @author Jack O'Quin
+ *  @author Piyush Khandelwal
+ */
+
+#ifndef VELODYNE_POINTCLOUD_POINT_TYPES_H
+#define VELODYNE_POINTCLOUD_POINT_TYPES_H
+
+#include <pcl/point_types.h>
+
+namespace velodyne_pointcloud
+{
+/** Euclidean Velodyne coordinate, including intensity and ring number. */
+struct PointXYZIR
+{
+  float x;                         ///< x coord
+  float y;                         ///< y coord
+  float z;                         ///< z coord
+  float intensity;                 ///< laser intensity reading
+  uint16_t ring;                   ///< laser ring number
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW  // ensure proper alignment
+}
+EIGEN_ALIGN16;
+
+};  // namespace velodyne_pointcloud
```

Comment:
Shouldn't we also define the PointXYZIRT in terms of a PCL structure so people can use it in the same way?

Reply:
Yes for melodic that makes sense. For noetic, I'll move this to the velodyne_pcl package.

Reply:
Sounds good.

Reply:
This item still needs to be done.